src/Servers/HttpSys/src/RequestProcessing/RequestStreamAsyncResult.cs

```diff
@@ -135,6 +135,8 @@ internal void Complete(int read, uint errorCode = UnsafeNclNativeMethods.ErrorCo
 
         internal void Fail(Exception ex)
         {
+            Dispose();
+            _requestStream.Abort();
```

Comment:
This was moved to further limit races between the Abort logic and the ReadAsync Task completion.

Reply:
Add a comment to the code saying this?